> `LOG(FATAL)` calls `std::abort`. So `return` after it should not be needed. Is there a compile failure causing these returns to be needed? If yes, can we actually try to use `std::abort`?

yes, on Windows debug MSVC (2019) build I get errors like these:
`C:\...\execroot\org_tensorflow\tensorflow\core\framework\device_base.cc(72) : error C4716: 'tensorflow::DeviceBase::name': must return a value`
`C:\...\execroot\org_tensorflow\tensorflow\core\framework\device_base.cc(68) : error C4716: 'tensorflow::DeviceBase::attributes': must return a value`

Apparently in debug mode MSVC compiler doesn't get that `LOG(FATAL, ...)` will finally call `abort()`...

I agree that using `std::abort()` in all cases, instead of returns, is nicer. I'll push an update to this PR.

--- a/tensorflow/core/kernels/depthtospace_op.cc
+++ b/tensorflow/core/kernels/depthtospace_op.cc
@@ -178,7 +178,7 @@ template <typename T>
 struct DepthToSpaceOpFunctor<CPUDevice, T, FORMAT_NCHW> {
   void operator()(const CPUDevice& d, typename TTypes<T, 4>::ConstTensor input,
                   int block_size, typename TTypes<T, 4>::Tensor output) {
-    LOG(FATAL) << "dummy implementation to make debug build compile";
+    LOG(FATAL) << "Trivial implementation to make debug build compile.";
   }
 };
 #endif

--- a/tensorflow/core/kernels/spacetodepth_op.cc
+++ b/tensorflow/core/kernels/spacetodepth_op.cc
@@ -194,7 +194,7 @@ template <typename T>
 struct SpaceToDepthOpFunctor<CPUDevice, T, FORMAT_NCHW> {
   void operator()(const CPUDevice& d, typename TTypes<T, 4>::ConstTensor input,
                   int block_size, typename TTypes<T, 4>::Tensor output) {
-    LOG(FATAL) << "dummy implementation to make debug build compile";
+    LOG(FATAL) << "Trivial implementation to make debug build compile.";
   }
 };
 #endif

